dna-as-probabilistic-program.md

- probabilistic programming languages
- structure => function && function => structure

establish the overall theme and tie each part back to how it reinforces this theme
- genetics is a program where loops and trees of behavior can be predictably triggered
  - so go through examples from epigenetics and explain how they reinforce/demostrate/utilize this idea

how to structure this series of articles?

article about predictive medicine and CYP450
- this article might be a good start for probabilistic theory
  - but how to measure this?
  - explain why the expression of genetics has a much greater effect
    than the encoding of alleles in the genes, usually

- hmmmmm, there was like one other example of probabilistic
  programming theory of metanomics
  - neural fingerprints

metanomics and probabilistic programming
- mapping out tissue networks in the human body using MRI (diffusion
  tensor imaging) + Yoga.
  - need to be able to map various spaces together
  - the yoga postures cause different blood flow patterns
  - statistical resolution should allow us to map out tissues by
    looking at networks of small veins and arteries down to
    capillaries
- this should revolutionize cancer treatments because you can tailor
  someone's treatment plan to target specific networks of tissues
  - you can know that if you take out a set of tissue subnetworks,
    that you have taken out all of the branches leading to them
  - if the cancer is confined to that area, then you can know that
    there's a much smaller chance of it coming back
    - you have annihilated the tissue 'feeding' the cancer
    - so therefore, there should be a much lower chance of it
      returning
- this is like a "connectome" for probabilistic programming
  - the connectome is the highest level of tissue interconnection, but
    is stratified itself, as it's a tree
  - the brain has a connectome and this can be observed by using
    diffusion tensor imaging and some learning
  - but the yoga in the methodology inserts some known level of
    variance into the system of veins/arteries for specific areas of
    the body
  - those areas get more blood and open up
    - the MRI images can be sewn together and with enough imaging
    - then a highly accurate model of the connectome for the brain and
      other organs can be constructed

part one: introduction:

- what is probabilistic programming
  - explain PP languages that are being explored
  - and why the research is being done (promising for machine learning & ai)
- explain what makes something a program
  - trees/loops
  - explain how it's hard for people to see how a set of probabilities
    - can lead to something with behaviors like a program
    - powersets of interactions, etc, etc
  - cells can even use metaprogramming
- list current models for biological behavior that this model circumscribes
  -
- what's the difference between current models and this paradigm for genetic behavior?
  - this model not only explains how genetics works in the nucleus, which directs operations of the cell
    - but also explains cytoplasmic interactions and cytoplasmic-nuclear interactions
      - that is, spatially distributed state extends beyond the nucleus
    - it also can help explain inter-cellular interactions
      - caused by chemical messengers (hormones & proteins)
      - or caused by more global change in state (pH, hydration)
      - or caused by more local change in state (changes across specific organs)
    - it can also explain inter-chromosome behaviors
  - current models seem limited to describing interactions between related genes
    - they don't describe how different chromosomes interact
    - they are overly simplistic in describing how protein products of genes interact
    - describing how controlling spaces within the cell and nucleus
      - are totally out of scope for current models
  - the probabilistic programming paradigm for genetic behavior is more complete
    - this makes it more complicated, but also simpler at the same time
    - basically, for each gene, protein, nucleoulus, transcriptome, cell, group of cells, organ, body, etc
      - there are a sets of functions that can be executed, each with a probability
        - executing the function changes the local state, which in the case of genetics is spatially distributed
    - looking at the complete picture while resolving the most minute details and interactions
      - is probably impossible, as you have to consider powersets of interactions and behaviors
    - but it should be possible to describe some behaviors and tendancies that could not be explained before

part two: spatially distributed state
- explain what I mean by spatially distributed state
- transciptomes and nucleolus
  - use Process/Program & OS metaphors to describe these


part ??: metaprogramming
- some cells modify their own genomes to augment behaviors



- DNA as a probabilistic program
  - differentiation of genomic programs and cell-types should be relatively flat
  - that is, there's a tree that represents the types of cells that cells can become
  - but, because of occam's razor, this tree should be flat because too much hierarchy induces too much complexity in the system
  - genetic programs modelable by markov graph?

- http://www.scientificamerican.com/article/human-genome-s-spirals-loops-and-globules-come-into-4-d-view-video/

An offhand, uninformed, speculative response to #1 - DNA that encodes mRNA should not be located directly adjacent to chromatin linkage sites. Junk DNA may also contain metadata that directs the folding of chromatin.

Junk DNA *may* also be expressed when an activation factor is not connected to properly transcribe a gene and thus, transcription begins at the wrong site. *If* this is the case, expression of some Junk DNA may form the basis of an if-then-else feedback loop that indicates to genetic programs that a particular gene is *not* being expressed. For more info, see this article: http://www.the-scientist.com/?articles.view/articleNo/43029/title/Silencing-Surprise/

probabilistic programming languages

other features of computer programs that must have a mirrored mechanism in genetic programs
- clocking mechanism (circadian rhythm and others)

distinctions between genetic programs and computer programs
- all programs are tree like structures
- however, because of occam's razor,
  - genetic program trees are likely very flat
  - and the dependency tree of differentiated cell types is likely very flat as well

implications of mitotic heritability
- genes that are turned on in one cell type are also turned on in dependent cell types
  - unless they are explicitly deactivated by the dependent cell type
  - therefore, genes expressed in stem cells are all active in dependent celltypes
    - unless explicity turned off
    - TET proteins remove DNA methylation, at specific points in development
      - TET proteins are very rarely expressed
        - supporting my hypothesis, at least for the methylation mechanism
        - however, methylation usually silences genes
      - primarily expressed in primordial germ cells and embrionic stem cells

because loosely packed chromatin is more available to molecules around it
- it should be easier to silence genes than it is to activate them
- that is, it's easier for epigenetic machinery to access histones of active genes
  - than inactive genes

because histone H1 winds up nucleosomes in an ordered fashion,
- since chromatin is wound up in a spiral formation,
  - some histone types should always be center-facing and some should be outward facing.
- therefore, some histone types should be more readily accessible to epigenetic machinery than others.
  - that is, when genes are activated, these histones will be be the first to be altered
  - which opens up the nucleosomes' more hidden histone types.


shape and location of chromosomes definite affect their function
- chromatin near the periphery is usually heterochromatin
- chromatin near the center is usually euchromatin

### blog post: in nucleus transcription factories,
- RNA Polymerase 2 is static and DNA is pulled through
  - this is sort of like a program executing in a linear fashion
- transcription factories are often packed with transcription factors
  - different transcription factories have variant levels of transcription factors
- coregulated genes share transcription factories
  - this is more efficient bc instead of requiring high levels of transcription factors everywhere
  - there is just needed
- transcription factors are like CPU cores? (or a computing process)
  - in that they control the execution of a program (DNA strand)
  - and contain their own environment (transcription factors)
  - maybe better computing analogy

what is the function of the transcription factory membrane?
- do nuclus transcription factories block entry of certain proteins?
  - would it react differently to lncRNA's than to proteins
- does the membrane 'lock in' transcription factors and enhancers on it's surface
  - doing so would enable the fast expression and re-expression of specific sections of the genome
  - in otherwords, it would enable a loop from reexecuting the same genes
    - and the buildup of a byproduct of the expression of these genes could dissipate the transcription factory
  	- providing the 'until' condition for a repeat-until loop

nucleolus seems like the process of the most basic DNA program
- the program that ensures ribosomes are created and mRNA can be transcribed
  - contains RNA Pol1, which transcribes genes for ribosomes
  - surrounded by RNA Pol3, which transcribes genes for tRNA, which translates codons to amino's
- the reason the nucleolus contains RNA P1 and is surrounded by RNA P3
  - is because the nucleolus is like the operating system process of DNA
  - and so all of this needs to be in one place
- the other transcription factories are like spawned processes of the operating system


x-inactivation requires threshold level of Rnf12
- with only one active X(in XX or XY), there is not enough Rnf12 to trigger the code for inactivation
- with two active X expressing Rnf12, there is sufficient Rnf12 so that P(inactivation) is close to one
- once the x-inactivation program runs, there is no longer enough Rnf12 around, so P(inactivation) is close to zero
- however in Rnf12-knockout, x-inactivation can still occur, but usually occurs much later

x-inactivation doesn't occur when x-chromosomes can't spatially pair
- i really don't think there is a 'counting' mechanism.
- Rnf12 is more like a binary mechanism -- X.count == 2 or X.count == 1
- if the two chromosomes are adjacent, how does the X chosen to activate protect itself against Xist?
  - where is Xist gene located in relation to the X chromosome pairing site
  - if the Xist genes are typically located opposite from the pairing site
    - this would protect the X chromosome that is chosen to be active from accidental inactivation

symmetry in histone arrangement may provide various kinds of "memory" for epigenetic processes
- first level of symmetry: histones adjacent to H1 face inward towards chromatin
  - in heterochromatin, these histone tails would be mostly inaccessible
  - in more active euchromatin, these histone tails may have some other function
- .....

categorize relationships between genes and the factors affecting their relationships
- e.g. are two functionally related genes close enough that their proximity affects their relationship
- e.g. are two genes activated by similar transcription factors?

early after fertilization, the fertilized egg has two pro-nuclei
- this allows for imprinting of parent-of-origin gene expression
  - is this the only time parent-of-origin gene expression can occur? probably not
- this implies genetic programs can run with one copy of each chromosome in each nucleus

epigenetic reprogramming immediately after fertilization
- both maternal and paternal genomes reach methylation low around the blastocyst stage
  - paternal genome is demethylized fast by hydroxylating the methyl groups, which destabilyzes the bonds
    - involves TET proteins
  - maternal genome is demethylized more slowly because DNMT1 is not expressed
  - however, what accounts for this difference in behavior?
  - is DNMT1 only suppressed in the maternal genome or both?
  - what triggers hydroxylation in paternal genome?
  - how long do the pro-nuclei remain separate?  only in the fertilized egg?

epigenetic reprogramming for primordial germ cell development
- in this case, paternal and maternal germ cells are cleared of methylation at about the same rate
  - but spermatogenesis remethylation proceeds slightly faster
  - oogenesis remethylation proceeds more slowly

however, repeats(IAP's) are generally silenced throughout epigenetic reprogramming during development
- this is because removing the methylation from these would lead them to be expressed
  - and expression of IAP's at the wrong time can cause detrimental problems
- sometimes, these repeat regions are briefly demethylized (sometimes around the blastocyst stage)
  - however, this is usually accompanied by expression of complimentary piRNA's which quickly re-establish methylation
  - ** good example of predictable programmatic behavior here.
    - expression of IAP repeats is predictably followed by the targeted expression of complementary piRNAs
    - how is this orchestrated?

genomically imprinted genes (aka parental imprinting, gametic imprinting)
- these genes have monoallelic gene expression (i.e. only one of two alleles is expressed)
  - but the expression of alleles occurs in a parent-of-origin restricted manner
  - only your maternal copy or only your paternal copy is expressed
- these imprinted genes undergo reprogramming during primordial germ cell development
  - but resist the initial reprogramming
- these gene sets are known to be critical for viability of embryo's

expression of imprinted genes is controlled by imprint control regions (ICR's)
- aka Differentially Methylated Domains/Regions (DMD/DMR)
- imprint is universally associated with DNA methylation at the ICR
  - e.g. paterally imprinted genes are marked with DNA methylation
    - however, this doesn't necessarily imply genes imprinted with methylation are always silenced
- ICR's express imprinted genes differently.  each locus is different
  - lncRNA's, enhancer/promotor blocking
- ICR methylation is established during primordial germ-cell development!
- protection from clearing of DNA methylation involves oocyte derived maternal effect proteins
  - these proteins are expressed in the egg and protect ICR's from the widespread demethylation
- in primordial germ cells, these ICR's are erased and then re-established

imprinted genes exist in clusters.  there are ~150 imprinted genes in humans and ~130 in mice.
- each cluster has it's own ICR.  and it's the ICR that controls the expression of the surrounding genes.
- imprinted gene expression is not always found in all tissues.
  - imprinted gene expression is common in placenta and brain.

** ICR resembles like an if-then-else.
- when ICR is marked, some genes will be expressed and others repressed.
  - refer to document for mouse geneome in video 4-3
- e.g. if the ICR is imprinted on the maternal chromosome,
  - this will result in the maternal expression of certain genes
  - since the ICR is not marked on the corresponding paternal chromosome
    - this results in the expression of another set of paternal genes within the same cluster

kcnq1 ICR (Chromosome 11) controls gene expression with the lncRNA kcnq1ot1
- maternal methylation of kcnq1 ICR silences kcnq1ot1
  - the lack of lncRNA expression allows the other kcnq1 genes to be expressed
  - on the paternal side, the expression of the lncRNA silences expression of other kcnq1 genes
- kcnq1ot1 silences in cis, recruiting G9a and PRC2 to perform H3K9me and H3K27me
  - kcnq1ot1 is transcribed in antisense to kcnq1
- however not all genes adjacent to kcnq1 are silenced
- cdkn1c is maternally expressed and is a growth inhibitor (tumor suppressor)

### blog post: in addition to being crucial for DNA replication,
- the duality of DNA is also crucial for the function of miRNA's, piRNA's and  lncRNA's
- evolutionarily, the duality of DNA promotes the similarity of long sequences of adjancent DNA
  - this promotes the function of miRNA's, piRNA's and lncRNA's
  - e.g. if large copy-and-paste mutation events occur, then similar regions may be created a short distance from each other
  - then it's more likely that the RNA complement of an mRNA can be expressed under certain circumstances
    - and this RNA complement can act as an miRNA, piRNA or lncRNA

### DNA transcription game
- similar to pacman where you travel on DNA strands, but involves a shifting level (3D?)
  - more focus on node-node connections instead of length of maze segments?
- running across certain regions will trigger certain functions
  - release enhancer/inhibitor proteins that follow you and neutralize enemies
  - or trigger strands of the level to connect temporarily in new ways
  - combinations of enhancer/inhibitor proteins can be used for AoE attacks

H19/igf2 cluster (Chromosome 11)
- H19 is a maternally expressed lncRNA, but it does not affect methylation
  - instead it's a miRNA resevoir
- m9n at the ICR instead blocks the CTCF from binding to the ICR
  - without CTCF, m9n spreads to the H19 promotor, deactivating it
  - CTCF is an insulator protein and insulates the igf2 gene from downstream enhancers
- when H19 is active, the downstream enhancers will upregulate H19
  - CTCF will block enhancers from binding to and acting on igf2
- when CTCF is insulating, H19 is deactivated by methylation
  - CTCF is not expressed and so enhancers bind to igf2
- it's thought that this action is modulated by chromatin looping
  - CTCF modulates the chromatin loop and so without it, enhancers are adjacent to H19
  - with CTCF, enhancers are adjacent to igf2

somehow, the act of transcription through a locus can deactivate a gene
- still not know exactly how the silencing is set up in these cases.
- not known if Ube3a-antisense or the act of transcribing Ube3a in reverse causes inactivation
- possible that the act of transcribing requires the loss of histone marks??

"Curiously, many more maternally-methylated than paternally-methylated DMRs/ICRs have been identified."
- this may be because it's easier to conserve methylation for maternally imprinted genes
- because of the maternal effect proteins, which are produced in the oocyte

intracisternal A-particles as mechanism for genetic metaprogramming
- these IAP's carry LTR's (long terminal repeats) on either end
- when inserted between exons for genes,
  - this presents the genome with an opportunity to alternativly express these genes
  - over time, this means cells can recombinate alternate expression of genes
    - e.g. Axin-fused mice with the IAP insert can express the protein Axon as both:
      - [...+exon6+exon7+...]
      - or [exon7+...] - which is the expression type that causes kinked tails in mice
    - this behavior can be modulated by epigenetic state
      - but otherwise the gene would be only expressed as [...+exon6,7+...]
      - because there's no mechanism to signal the alternate expression
- IAP's cause lots of problems but also present our cells with a myriad of expressions for the same gene
  - it's like metaprogramming with the statements of a lisp function
  - you can create a macro that deconstructs a lisp function into a list of functions
    - you can now combinate through this list of functions to rearrange elements or insert/modify/remove elements
  - by attempting to run the different variations of function recombinations
    - along with tests to ensure the behavior of these variations satisfies some notion of equality 4 the expected behavior
      - category theory FTW
    - then you can create new behaviors for your program by identifying the successful recombinant variations
  - but in the case of the IAP's, survival of the fittest is your 'test' for successful variations of gene expression
    - and each generation can only recombine a gene's exons a single time
      - it's not very likely for IAP's to be insert itself more than once between three or more exons of the same gene
      - fortunately, for evolution, this simplifies things, as the new gene can only be expressed a limited number of ways
- IAP's greatly accelerate evolution
  - they can split genes into multiple exons, which can be recombined to become more functionally useful
    - that is, if a gene contains a region that is a receptor for an activation site
- can the recombinant function method described above be used for artificial intelligence
- can the IAP mechanism of insertion of variations of genetic-metadata be used with bytecode
  - in otherwords, take a region of bytecode for a function
    - and randomly insert various segments of program-altering bytecode
    - e.g. ([begin & instructions] + {new-end + new-instructions + new-begin} + [instructions + end])
      - creates a new callable function
      - dealing with this machine code would be a serious mind fuck
  - similar techniques are probably already widely used for compiler optimization
    - but can these techniques be used for self-modifying programs
    - that intentionally destructure/restructure their machine code to create new behaviors
    - while satisfying dependencies and objectives
- and apparently the recombinant behavior i'm describing is very similar to concatenative programming languages
  - such as Joy and Factor
  - concatenative.org/wiki/view/Concatenative%20language/Meta-programming


role of protamine in transgenerational epigenetic inheritence
- sperm doesn't have chromatin. instead it has protamine, which retains some histone marks.
- how could epigenetic information be retained through reproduction, conception and through the blastocyst stage
  - the blastocyst stage resets methylation


paramutation in mice: kit protein that causes mice to fully develop black coat of fur
- when injecting wild-type (black) mouse eggs with kit protein mRNA from a spotted mouse
  - this sometimes causes the mouse that's born to become spotted
  - but not usually if you inject mRNA from
    - very rarely, injecting this mRNA from a black mouse will cause a spotted coat
    - could this be due to the similarity for the Kit protein mRNA between black & spotted mice?
    - when mRNA is exact match, it triggers process
    - when mRNA is inexact match, it rarely triggers the process
    - this might mean DNA's complementary nature is at play


miRNA's for piRNA's?
- they would cancel out piRNA's to end the execution of piRNA's

miRNA treatments for diseases involving imprint control regions
- the miRNA would could cancel out the misexpressed allele
- would it be specific enough
  - if not, could it be used to just generally tune down the expression of those genes

lncRNA's as genetic state-machines
- it'd be interesting to identify lncRNA's which can transition their shape from one form to another
  - each triggering some kind of functionality
    - lncRNA1 transitions (1) A -> B (2) B -> C (3) C -> B (4) C -> D
- this would make them like a genetic state machine
  - they could have certain specific functionalities which are only enabled while in certain forms
  - this would be disadvantageous if they became "stuck" in some forms
    - that is, if they are needed in each cell of a cell type
      - if they are used only occasionally to trigger some permananent state change, this would be ok
      - otherwise, they would trigger the production of too much waste in a nucleus
