---
title: "Metanomics & Probabilistic Programming: Epigenomic Metaprogramming"
categories: blog
tags: "genomics epigenetics"
headline: ""
author:
  name: "David Conner"
excerpt: ""
---

- is there a good example from the epigenetics class?
- again, another good example?

- reconstruction of realistic epigenetic marker values from fossils up
  to 5,000 years old?
  - it MAY be possible. basically, we'd have to have enough well
    preserved (ice-frozen or peat-bog preserved) to have a ton of data
    to sample from, along with a common map for expected values on
    genes.
  - so, basically the idea is to compare epigenetic data assumed to be
    highly erroneous with a methylation epigenome assembled today.

    - if it's possible there's mostly no way it'd work for histone
      values. i'm not sure if those can be sampled cheaply &
      non-invasively today
    - but WHAT IF it's possible for *some* DNA methylation.
    - it should be possible, especially in context of some epigenomic
      maps for tissue.
      - you can compare the methylation epigenome for different tissue
        types today with the methylation scanned in from various
        tissue types from some well-preserved sample like the iceman
      - because we know what the expected range of variation in tissue
        types is b/w

  - IFF it's possible!! (it's a big iff) to sample epigenomic data
    from for well-preserved tissue samples *with nearly any degree of
    accuracy*, then this should give us INVALUABLE insight into the
    rate of evolution for SOME epigenomic processes.
    - we might be able to see, with genomic & epigenomic variation,
      how humanity EPIGENETICALLY responded to developments such as
      agriculture & the development of culture
      - this *could* perhaps stretch back to well-preserved samples
        from 100, 500, 2000 and 5000 years ago!
        - it could also just be flat out impossible, depending on how
          methylation (and DNA!) degrades over time
        - if we can't sequence the genome of the Iceman for whatever
          reason, the it's impossible... *but if it's not!* then it
          may be that we'll uncover a major clue in the development of
          humanitys most recent evolutionary changes
    - however, i am less interested in humanity's evolutionary changes
      in the past 3000 years and much more interested in understanding
      how humanity responded epigenetically with respect to the
      development of agriculture, commerce, writing, architecture,
      etc.
      - but if you look at "extinct" ice age animals over the past
        1,000,000 years, like wooly mammoths, IMO many of these
        animals are just like extreme hybridizations of animals which
        exist today.
        - this is probably a dumb statement on my part, but i'm pretty
          sure wooly mammoths and elephants are so closely related so
          that the genetic material might be closer than that of
          chimps & humans. it'd be a very simple question to answer.
        - there may be some of these extinct ice age animals whose
          epigenetic markers quickly responded to the change in
          environment over 5-10 generations during periods of extreme
          environment shift when an ice age began/ended
      - what is interesting to me is determining how the end of the
        last ice age affected the epigenome of man

    - (unfortunately, there could be some darker aspects to this
      research ... there are some questions about human history and
      human evolution that are most definitely better left unanswered)
        - DONT EVEN THINK ABOUT IT. THESE ARE BAD QUESTIONS TO ASK.
        - and the capacity for an epigenome to respond may imply that
          humans can adapt to sociocultural changes within one to
          three years.
        - but you know what would be great? is if people didn't think
          about human's destiny or ability being predetermined by
          genetics, regardless of how true or false it is. i want to
          believe that every person makes their own destiny,
          especially if that belief would fly in the face of science.
          - if the science demonstrates such conclusions, then the
            mature response is to ignore them to spite science with
            the idealistic view that every man makes himself over the
            course of his life.
          - and, in fact, this is shown to be true. regardless of the
            state of your epigenetic markers at birth (barring genetic
            diseases, etc) the activities your partake & your diet &
            your environment shape how your epigenome are expressed
            over the course of your life.
            - the truth is, if you hit the books for the first 20
              years of your life, it doesn't matter what your genome &
              epigenome reported, you're going to train your brain to
              become smarter. i get the feeling that these changes are
              passed down to our children.
            - in other words, epigenetically, you are what you eat and
              you are what you do. our bodies and organs and brain all
              have the capacity to change the functioning of our
              genetic program.
          - it would really hurt me, regardless if i had anything to
            do with this science or not, to find out that some
            genetic/epigenetic science was used as rhetoric fodder for
            stupid racist bullshit.
            - and actually, if you look forward 20 years, we'll move
              past race & color and everything with genetic
              engineering...
              - anime got it right. all those people in neo-tokyo in
                2050 really do have !@#$'ing purple hair and orange
                eyes!
          - my point is twofold:
            - (1) it's stupid to leverage this science to support
              racist bullshit. ain't nobody got the time for that
              dumbass shit. we all need ALL OF HUMANITY to come
              together because our goddamn house is on fire.
            - (2) if you're so hung up on justifying racism with
              nouveau genomic/epigenomic science, then guess what?! in
              20-30 years, we will be free to choose our own
              identities *to a degree never before known by humanity*,
              so your dumbass intolerant shit will be outmoded.
          - that's not to say that racial/ethnic issues will become
            unimportant, but in the near future we will leverage more
            control over our personal identity and the identity of our
            children
            - and there are plenty of ethical concerns for genetic
              engineering. think about it like this:
              - while genetically engineering our children could
                eradicate diseases like huntington's ...
              - picking designer characteristics for your children is
                a bit like a tattoo that might last multiple
                generations
                - now your children's children might have purple hair,
                  yada yada...

### Does that guy really have the "Y" score? ... lulz

- but basically, the question sparked off this idea for epigenomic
  analysis of well-preserved specimens was my puerile obsession with
  Genghis Khan's 800 children... lulz (i should really, really STFU
  sometimes)
  - but ... how do we know that guy really has the high score? LMAO
  - and honestly, this isn't funny ... but it kinda is.
  - apparently that guy had 800 kids and there are now like hundreds
    of millions of descendents of his Y chromosome variant that were
    replicated over 800 years
- but are there other Y chromosome variants from the last 5000 years
  which were more prevalent than Genghis Khans?
- honestly, this is incredibly immature, but i can't help it
  sometimes. that factoid about Genghis Khan is hilarious.

### Some cells modify their own genomes to augment behaviors

- some cells modify their own genomes to augment behaviors
  - genomic process for copy & pasted genes
    - is there a genomic mechnanism for preventing bad writes?
    - some writes would have almost no effect on genetic expression of
      critical factors
    - some writes would be disastrous for that generation or the
      subsequent generation
    - some writes would be disastrous for all subsequent generations
      of cells
- or modify the available mRNA & their rates of transcription into
  protein products
  - miRNA, piRNA?
    - these ramp up or inhibit the available mRNA's that are to be
      produced into protein products
  - colon/prostate/etc(??? verify!) cancers can be caused by
    copy-and-paste-genes (what are these called again?)
    - basically that cell type needs lots and lots of copies of that
      gene, so it triggers the gene itself to be randomly copied
      elsewhere in the genome and these genomic (not epigenomic)
      changes are passed to daughter cells

establish the overall theme and tie each part back to how it
reinforces this theme
- genetics is a program where loops and trees of behavior can be
  predictably triggered
  - so go through examples from epigenetics and explain how they
    reinforce/demostrate/utilize this idea

how to structure this series of articles?

article about predictive medicine and CYP450
- this article might be a good start for probabilistic theory
  - but how to measure this?
  - explain why the expression of genetics has a much greater effect
    than the encoding of alleles in the genes, usually

- hmmmmm, there was like one other example of probabilistic
  programming theory of metanomics
  - neural fingerprints


# Misc Bioinformatics Ideas

### Protein Folding

#### from 2014

- {all of this is done, over and over in research. some of it is
  fairly recent, 2005-2015ish}
- protein folding algorithms that make inferences based on mutations'
  effects on protein structure
  - {there are certainly stochastic methods, which can be combined
    with *ab initio* folding algorithms}
    - the question here is about evolutionarily conserved mutations,
      identifying mutations that we see from the evolutionary record
      and from sequencing genomes of species and whether inferences
      can be made about the mutations that could have happened and
      what happened.
    - so, it can be assumed that some mutations, particularly those
      resulting in amino acid changes, result in aminos that are
      more or less likely to change the structure of features
      - i.e. features like alpha/beta helicies
  - if one takes this "ab initio + stochastic" approach then, tests
    protein folding ab-initio for protein variants with amino
    mutations that cause structure to diverge, then this approach
    tells you what is likely to be conserved and what is not.
    - so, protein structure needs to "anticipate" rates & distances of
      evolution. structure needs needs some "wiggle room" to preserve
      structure & form in spite of inevitable mutation.
    - so, it's interesting to see how one can make amino mutations
      and see how that changes form/function to see what form is
      conserved despite mutations.
    - protein folding is complicated. depending on how you process a
      strand of aminos, there are divergent paths of folding that
      emerge. so, this allows one to understand which folding paths
      would be conserved by amino mutations. the set of folding paths
      most usually conserved is probably a good bet.

- given many samples of DNA for some particular coding regions,
  identify mutations and a lineage of genotypes. (this is done, over
  and over in research)
  - for each mutation, correlate the mutated amino acid's change in
    function to identify negative mutations
    - compare the functional changes for amino-mutations with a
      functional understanding of that local region and the adjacent
      aminos.
  - this is basically a functional understanding of the evolutionary
    lineage of genotypes.
    - the preservation of some genotypes tells us which mutations were
      not disadvantageous.

- analyze introns/exons for local regions that seem to mutate more
  frequently (particularly introns)
  - correlate that with nucleotide counts (since some nucleotides have
    a greater capacity for mutation, similar to how some amino acids
    have varying mutation/preservation rates)
  - this is already done, but i'm not sure there's enough
    bioinformatics research on identify how introns affect behavior in
    the nucleus.
  - so this:
    - (1) helps us understand how introns affect genetic expression
      and RNA polymerase behavior
    - (2) and potentially identify genes where the splicing of
      introns/exons is non-standard in some way. introns should mostly
      not show strongly conserved regions.
      - but mRNA has its own idiosyncractic folding and, perhaps, this
        folding matters sometimes.

#### After 2014?

- identify differences b/w human & chimp DNA to datamine those
  regions .... for ... aliens
  - try to find regions of Human DNA where, genetically or
    epigenetically, defenses against retroviruses were lost or
    replaced
